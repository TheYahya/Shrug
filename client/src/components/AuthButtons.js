import { connect } from 'react-redux';
import React, { Component } from 'react'
import { GoogleAuthorize } from 'react-google-authorize';
import { startLogin } from '../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faGoogle } from '@fortawesome/free-brands-svg-icons'

const GOOGLE_CLIENT_ID = process.env.GOOGLE_CLIENT_ID

class AuthButtons extends React.Component { 
  responseGoogle = (response) => {
    let accessToken = response.access_token
    this.props.startLogin(accessToken)
  }
  render() {
    return (
      <GoogleAuthorize
      className="btn btn-outline-danger"
      clientId={GOOGLE_CLIENT_ID}
      buttonText={<span><FontAwesomeIcon  icon={faGoogle} /> Login with Google</span>}
      onSuccess={this.responseGoogle}
      onFailure={this.responseGoogle}
    />
    )
  }
}


const mapDispatchToProps = (dispatch) => ({
  startLogin: (accessToken) => dispatch(startLogin(accessToken))
});

export default connect(undefined, mapDispatchToProps)(AuthButtons);
