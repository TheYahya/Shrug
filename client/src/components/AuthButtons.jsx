import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { GoogleLogin } from 'react-google-login';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGoogle } from '@fortawesome/free-brands-svg-icons';
import { startLogin } from '../actions/auth';

const { GOOGLE_CLIENT_ID } = process.env;

const AuthButtons = ({ login }) => {
  const responseGoogle = (response) => {
    login(response.accessToken);
  }

  return (
    <GoogleLogin
      clientId={GOOGLE_CLIENT_ID}
      buttonText="Login with Google"
      onSuccess={responseGoogle}
      onFailure={responseGoogle}
      cookiePolicy={'single_host_origin'}
    />
  );
};

const mapDispatchToProps = (dispatch) => ({
  login: (accessToken) => dispatch(startLogin(accessToken)),
});

AuthButtons.propTypes = {
  login: PropTypes.func.isRequired,
};

export default connect(undefined, mapDispatchToProps)(AuthButtons);
