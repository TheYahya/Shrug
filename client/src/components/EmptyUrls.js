import React from 'react';

const EmptyUrls = () => {
  return (
    <div className="recent-urls-empty text-center">
      <h4>No URLs shortened yet!</h4> 
      <img src="/images/empty-results.png" />
    </div>
  )
}

export default EmptyUrls;