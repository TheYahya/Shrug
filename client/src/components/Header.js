import React from 'react';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSignOutAlt } from '@fortawesome/free-solid-svg-icons';
import { Link } from 'react-router-dom';

class Header extends React.Component {
  render() { 
    const auth = this.props.auth.isAuthenticated === true ? <button className="btn btn-outline-secondary btn-sm" onClick={() => { this.props.startLogout() }}><FontAwesomeIcon icon={faSignOutAlt}/> Logout</button> : '';
    const userEmail = <span className="badge badge-light">{this.props.auth.email}</span>
    return (
      <header> 
        <nav className="navbar navbar-light">
          <span className="navbar-text">
            <Link className="navbar-brand" to="/">
              <img src="/images/shrug-ir.png" />
            </Link>
            <a className="navbar-text__link" href="https://github.com/theyahya/shrug" target="_blank">Github</a>
          </span>
          <div className="nav-right">
          { this.props.auth.isAuthenticated && userEmail }
          &nbsp;
          { auth }
          </div>
        </nav>
      </header>
    )
  }
} 

const mapSteteToProps = (state) => {
  return {
    auth: state.auth
  };
};

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(mapSteteToProps, mapDispatchToProps)(Header);
