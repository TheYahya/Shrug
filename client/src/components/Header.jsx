import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSignOutAlt } from '@fortawesome/free-solid-svg-icons';
import {
  Container,
  Navbar,
  Nav,
  Button,
  Badge,
} from 'react-bootstrap';
import { startLogout } from '../actions/auth';

const Header = ({ auth, logout }) => {
  const { isAuthenticated, email } = auth;

  return (
    <Container
      as="header"
      fluid
    >
      <Navbar
        bg="transparent"
        expand="md"
        collapseOnSelect
      >
        <Navbar.Brand>
          <Link to="/">
            <img alt="logo" src="/images/shrug-ir.png" />
          </Link>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <div className="d-flex w-100">
            <Nav className="mr-auto">
              <Nav.Link
                href="https://github.com/theyahya/shrug"
                target="blank"
              >
                Github
              </Nav.Link>
            </Nav>
            <Navbar.Text>
              {isAuthenticated && (
                <Badge
                  className="mr-1 bg-light"
                >
                  {email}
                </Badge>
              )}
              {
                isAuthenticated && (
                  <Button
                    onClick={logout}
                    variant="outline-secondary"
                    size="sm"
                  >
                    <FontAwesomeIcon icon={faSignOutAlt} />
                    Logout
                  </Button>
                )
              }
            </Navbar.Text>
          </div>
        </Navbar.Collapse>
      </Navbar>
    </Container>
  );
};

const mapSteteToProps = (state) => ({
  auth: state.auth,
});

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(startLogout()),
});

Header.propTypes = {
  auth: PropTypes.instanceOf(Object).isRequired,
  logout: PropTypes.func.isRequired,
};

export default connect(mapSteteToProps, mapDispatchToProps)(Header);
