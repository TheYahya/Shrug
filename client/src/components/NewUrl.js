import React from 'react';
import { connect } from 'react-redux';
import { startAddUrl } from '../actions/urls';
import AuthButtons from './AuthButtons'

const BASE_URL = process.env.API_BASE_URL

class NewUrl extends React.Component {
  state = {
    link: '',
    address: '',
    showOptions: false
  }
  onUrlChange = (e) => {
    const link = e.target.value;
    this.setState(() => ({ link }));
  }
  onAddressChange = (e) => {
    const address = e.target.value;
    this.setState(() => ({ address }));
  }
  onSubmit = (e) => { 
    e.preventDefault(); 
    const url = {
      link: this.state.link,
      short_code: this.state.address
    }
    this.props.startAddUrl(url)
    this.setState(() => ({ link: "", address: "" }));
  }
  onChangeOptions = (e) => {
    const showOptions = e.target.checked;
    this.setState(() => ({
      link: this.state.link,
      showOptions: showOptions
    }))
  }
  render() {
    if (! this.props.auth.isAuthenticated) {
      return (
        <div className="new-link text-center">
          <p>
            <img src="/images/shrug-ir.png" />
          </p>
          <h1>Shrink your urls with a <code>shrug</code></h1>
          <br/>
          <AuthButtons />
        </div>
        )
    }
    return (
      <div className="new-link">
        <form onSubmit={this.onSubmit}>
          <div className="form-row"> 
              <div className="form-group col-12">
                <input type="text" id="url" className="form-control" onChange={this.onUrlChange} value={this.state.link} placeholder="Your loooooong URL" autoComplete="off" autoFocus/>
              </div>
            </div>
            <div className="form-row">
              <div className="form-group col-12">
                <input id="more-options-input" name="more-options" onChange={this.onChangeOptions} type="checkbox"/> Custom url
              </div>
            </div>
            <div className="more-options" style={{display: this.state.showOptions ? 'block' : 'none'}}>
              <div className="form-row">
                <div className="form-group col-6">
                  <label>{BASE_URL}/</label>
                  <input type="text" id="custom-address" className="form-control form-control-sm" onChange={this.onAddressChange} value={this.state.address} placeholder="Custom url (optional)" autoComplete="off"/>
                </div> 
              </div> 
            </div>
          <small>Press <code>Enter</code> to shorten the URL.</small>
          <p><code>{ this.props.urls.addUrlError }</code></p>
          <input type="submit" style={{display: 'none'}} />
        </form>
      </div>
    )
  }
}
const mapSteteToProps = (state) => {
  return {
    auth: state.auth,
    urls: state.urls
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddUrl: (url) => dispatch(startAddUrl(url))
});

export default connect(mapSteteToProps, mapDispatchToProps)(NewUrl);
