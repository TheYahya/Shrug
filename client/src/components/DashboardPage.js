import React from 'react';
import NewUrl from './NewUrl';
import RecentUrls from './RecentUrls';
import { connect } from 'react-redux';
import QRCodeModal from './QRCodeModal';

const DashboardPage = (props) => {
  if (props.auth.isAuthenticated) {
    return (
        <div>
          <NewUrl />
          <RecentUrls /> 
          <QRCodeModal />
        </div>
    )
  }
  return (
    <div>
      <NewUrl />
    </div>
  )
}

const mapSteteToProps = (state) => {
  return {
    auth: state.auth
  };
};

export default connect(mapSteteToProps)(DashboardPage);