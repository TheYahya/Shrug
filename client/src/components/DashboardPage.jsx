import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import QRCodeModal from './QRCodeModal';
import NewUrl from './NewUrl';
import RecentUrls from './RecentUrls';

const DashboardPage = ({ auth }) => {
  if (auth.isAuthenticated) {
    return (
      <>
        <NewUrl />
        <RecentUrls />
        <QRCodeModal />
      </>
    );
  }
  return <NewUrl />;
};

const mapSteteToProps = (state) => ({
  auth: state.auth,
});

DashboardPage.propTypes = {
  auth: PropTypes.instanceOf(Object).isRequired,
};

export default connect(mapSteteToProps)(DashboardPage);
