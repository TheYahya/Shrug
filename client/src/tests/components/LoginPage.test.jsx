import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});
