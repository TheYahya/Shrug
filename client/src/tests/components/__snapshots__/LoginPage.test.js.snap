// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate
    </h1>
    <p>
      Tag line for app.
    </p>
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
